ntske.go

```diff
@@ -45,15 +44,15 @@ const (
 )
 
 func (s *Session) performKeyExchange() error {
-	tlsConfig := s.tlsConfig
-	if s.tlsConfig == nil {
-		tlsConfig = &tls.Config{}
+	dialer := s.options.Dialer
+	if dialer == nil {
+		dialer = func(ntskeAddress string, tlsConfig *tls.Config) (*tls.Conn, error) {
+			dialer := &net.Dialer{Timeout: s.options.Timeout}
+			return tls.DialWithDialer(dialer, "tcp", ntskeAddress, tlsConfig)
```

Comment:
nit: I know this works as the second `dialer` is scoped within the function, but it might be slightly easier to read if the variable had another name

```suggestion
			d := &net.Dialer{Timeout: s.options.Timeout}
			return tls.DialWithDialer(d, "tcp", ntskeAddress, tlsConfig)
```